plugins/module_utils/_api_data.py

```diff
@@ -1329,7 +1335,7 @@ def join_path(path):
     ),
     ('ip', 'dns', 'static'): APIData(
         fully_understood=True,
-        stratify_keys=('name', ),
+        require_one_of=('name', 'regexp'),
```

Comment:
This should still use something similar to `stratify_keys`. (Determining whether an entry should be updated or a new entry should be added should be done based on `name` and `regex`.)

I'm not 100% sure yet how this should look like, but I think this will be a larger change.

Reply:
If I understood correctly, the problem with using either of them like that (in the DNS case) is that they are not unique. The pair of "name" or "regexp" together with "address" is unique, but it's perfectly valid to have multiple records with the same name but a different address (DNS round robin).

This gets further complicated by other types of records, such as CNAMEs which don't have address at all. I understand that we want diff/check_mode to be accurate, but trying to parse all possible scenarios with DNS is gonna be hard and personally I don't think I have enough knowledge to cover all of them.

Reply:
Ah right, I was only thinking of `A` records (and even then it's not unique, as you pointed out). Not having `stratify_keys` is the correct way to go, then. This is a potentially breaking change for some users though, so this should be mentioned in more detail in the changelog.